server/src/test/java/org/apache/druid/segment/realtime/appenderator/BatchAppenderatorTest.java

```diff
@@ -913,6 +939,10 @@ static InputRow createInputRow(String ts, String dim, Object met)
     );
   }
 
+  private void waitForPersists() throws InterruptedException
+  {
+    Thread.sleep(500);
```

Comment:
Sleeps are bad. These will make the unit testing slower. Also, I bet all the sleeps you added will make these tests quite flaky.

Reply:
The use case here is somehow wait for a concurrent task to check on its effect, it seems like a legitimate use for sleep. Let me know what you are thinking that could be better.

Reply:
oh I see your suggestion below...